src/sentry/middleware/integrations/integration_proxy.py

```diff
@@ -0,0 +1,141 @@
+            return False
+
+        is_valid = verify_subnet_signature(
+            timestamp=timestamp, request_body=request.body, provided_signature=signature
+        )
+        if not is_valid:
+            logger.error("invalid_sender_signature", extra=log_extra)
+
+        return is_valid
+
+    def _validate_request(self, request: Request) -> bool:
+        """
+        Returns True if a client could be generated from the request
+        """
+        from sentry.shared_integrations.client.proxy import IntegrationProxyClient
+
+        self.proxy_path = request.path[len(PROXY_ADDRESS) :]
+        self.headers = clean_proxy_headers(request.headers)
+
+        log_extra = {"path": self.proxy_path}
+
+        # Get the organization integration
+        org_integration_id = self.headers.pop(f"{PROXY_OI_HEADER}", None)
+        if org_integration_id is None:
+            logger.info("missing_org_integration", extra=log_extra)
+            return False
+        log_extra["org_integration_id"] = org_integration_id
+
+        self.org_integration = OrganizationIntegration.objects.filter(
+            id=org_integration_id,
+            status=ObjectStatus.ACTIVE,
+        ).first()
+        if self.org_integration is None:
+            logger.info("invalid_org_integration", extra=log_extra)
+            return False
+        log_extra["integration_id"] = self.org_integration.integration_id
+
+        # Get the integration
+        self.integration = Integration.objects.filter(
+            status=ObjectStatus.ACTIVE,
+            id=self.org_integration.integration_id,
+        ).first()
+        if self.integration is None:
+            logger.info("invalid_integration", extra=log_extra)
+            return False
+
+        # Get the integration client
+        installation = self.integration.get_installation(
+            organization_id=self.org_integration.organization_id
+        )
+        self.client: IntegrationProxyClient = installation.get_client()
+        client_type = type(self.client)
+        log_extra["client_type"] = client_type
+        if not issubclass(client_type, IntegrationProxyClient):
+            logger.info("invalid_client", extra=log_extra)
+            return False
+
+        return True
+
+    def _should_operate(self, request: Request) -> bool:
+        """
+        Returns True if the business logic of this integration should run.
+        """
+        is_correct_silo = SiloMode.get_current_mode() == SiloMode.CONTROL
+        is_proxy = request.path.startswith(PROXY_ADDRESS)
+        if not is_correct_silo and not is_proxy:
+            # Avoid more expensive checks if we can
+            return False
+
+        is_valid_sender = self._validate_sender(request=request)
+        is_valid_request = self._validate_request(request=request)
+        return is_valid_sender and is_valid_request
+
+    def __call__(self, request: Request):
+        if not self._should_operate(request):
```

Comment:
That's a good call, I'll try a refactor after addressing your comments to validate this works.

Reply:
It was doable, I just had to make the URL pattern a catchall. This will also let us share an authentication scheme with whatever we do for the RPC later on 👍